File: io.openems.edge.controller.api.modbus/src/io/openems/edge/controller/api/modbus/jsonrpc/GetModbusProtocolRequest.java
package io.openems.edge.controller.api.modbus.jsonrpc;

import com.google.gson.JsonObject;

import io.openems.common.jsonrpc.base.JsonrpcRequest;

/**
 * Wraps a JSON-RPC Request to query the Modbus Protocol from Modbus/TCP
 * Api-Controller.
 *
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "method": "getModbusProtocol",
 *   "params": {}
 * }
 * </pre>
 */
public class GetModbusProtocolRequest extends JsonrpcRequest {

	public static final String METHOD = "getModbusProtocol";

	public GetModbusProtocolRequest() {
		super(METHOD);
	}

	private GetModbusProtocolRequest(JsonrpcRequest request) {
		super(request, METHOD);
	}

	@Override
	public JsonObject getParams() {
		return new JsonObject();
	}

}

File: ui/src/app/edge/settings/profile/modbusapi/getModbusProtocolRequest.ts
import { JsonrpcRequest } from '../../../../shared/jsonrpc/base';

/**
 * Wraps a JSON-RPC Request to query the Modbus Protocol from Modbus/TCP
 * Api-Controller
 * 
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "method": "getModbusProtocol",
 *   "params": {}
 * }
 * </pre>
 */
export class GetModbusProtocolRequest extends JsonrpcRequest {

    private static METHOD: string = "getModbusProtocol";

    public constructor(
    ) {
        super(GetModbusProtocolRequest.METHOD, {});
    }

}